### ISSUE TITLE
Convert pure pytorch code to ignite 

### ISSUE THREAD

**Issue by Priyansi** [pytorch-ignite/examples#14]
Following on the discussion with @vfdev-5 , this new how-to guide should be abstract and:
> The purpose is to show explicitly how to convert pure pytorch code to ignite and explain what we gain by that

Topics to cover:
- Training loop
- Validation loop
- Metrics
- Checkpoints
- LR Scheduler
